test/e2e_node/mirror_pod_grace_period_test.go

```diff
@@ -53,36 +56,61 @@ var _ = SIGDescribe("MirrorPodWithGracePeriod", func() {
 
 		ginkgo.It("mirror pod termination should satisfy grace period when static pod is deleted [NodeConformance]", func() {
 			ginkgo.By("get mirror pod uid")
-			_, err := f.ClientSet.CoreV1().Pods(ns).Get(context.TODO(), mirrorPodName, metav1.GetOptions{})
+			pod, err := f.ClientSet.CoreV1().Pods(ns).Get(context.TODO(), mirrorPodName, metav1.GetOptions{})
 			framework.ExpectNoError(err)
-
-			start := time.Now()
+			uid := pod.UID
 
 			ginkgo.By("delete the static pod")
 			file := staticPodPath(podPath, staticPodName, ns)
 			framework.Logf("deleting static pod manifest %q", file)
 			err = os.Remove(file)
 			framework.ExpectNoError(err)
 
-			for {
-				if time.Now().Sub(start).Seconds() > 19 {
-					break
-				}
-				pod, err := f.ClientSet.CoreV1().Pods(ns).Get(context.TODO(), mirrorPodName, metav1.GetOptions{})
-				framework.ExpectNoError(err)
-				if pod.Status.Phase != v1.PodRunning {
-					framework.Failf("expected the mirror pod %q to be running, got %q", mirrorPodName, pod.Status.Phase)
-				}
-				// have some pause in between the API server queries to avoid throttling
-				time.Sleep(time.Duration(200) * time.Millisecond)
-			}
+			ginkgo.By("wait for the mirror pod to be running for grace period")
+			gomega.Consistently(func() error {
```

Comment:
Shouldn't this be Eventually if we're waiting? Consistently checks that it's true every time and will fail on the first false if it's not.